op-devstack/dsl/bridge.go

```diff
@@ -0,0 +1,489 @@
+	gameTypes "github.com/ethereum-optimism/optimism/op-challenger/game/types"
+	"github.com/ethereum-optimism/optimism/op-devstack/devtest"
+	"github.com/ethereum-optimism/optimism/op-devstack/stack/match"
+	nodebindings "github.com/ethereum-optimism/optimism/op-node/bindings"
+	"github.com/ethereum-optimism/optimism/op-node/rollup"
+	"github.com/ethereum-optimism/optimism/op-node/rollup/derive"
+	"github.com/ethereum-optimism/optimism/op-node/withdrawals"
+	"github.com/ethereum-optimism/optimism/op-service/apis"
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum-optimism/optimism/op-service/predeploys"
+	"github.com/ethereum-optimism/optimism/op-service/txintent/bindings"
+	"github.com/ethereum-optimism/optimism/op-service/txintent/contractio"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient/gethclient"
+	"github.com/holiman/uint256"
+	"github.com/stretchr/testify/require"
+)
+
+// ProvenWithdrawalParameters is the set of parameters to pass to the ProveWithdrawalTransaction
+// and FinalizeWithdrawalTransaction functions
+type ProvenWithdrawalParameters struct {
+	Nonce              *big.Int
+	Sender             common.Address
+	Target             common.Address
+	Value              *big.Int
+	GasLimit           *big.Int
+	DisputeGameAddress common.Address
+	DisputeGameIndex   *big.Int
+	Data               []byte
+	SuperRootProof     *bindings.SuperRootProof // Only set for games using super roots
+	OutputRootIndex    *big.Int                 // Only set for games using super roots
+	OutputRootProof    bindings.OutputRootProof
+	WithdrawalProof    [][]byte // List of trie nodes to prove L2 storage
+}
+
+type StandardBridge struct {
+	commonImpl
+	l1PortalAddr        common.Address
+	l1Portal            bindings.OptimismPortal2
+	l2tol1MessagePasser bindings.L2ToL1MessagePasser
+	disputeGameFactory  bindings.DisputeGameFactory
+	rollupCfg           *rollup.Config
+
+	l1Client         apis.EthClient
+	l2Client         apis.EthClient
+	supervisorClient apis.SupervisorQueryAPI
+}
+
+func NewStandardBridge(t devtest.T, l2Network *L2Network, supervisor *Supervisor) *StandardBridge {
+	l1Client := l2Network.inner.L1().L1ELNode(match.FirstL1EL).EthClient()
+	l1PortalAddr := l2Network.inner.RollupConfig().DepositContractAddress
+	l1Portal := bindings.NewBindings[bindings.OptimismPortal2](
+		bindings.WithClient(l1Client),
+		bindings.WithTo(l1PortalAddr),
+		bindings.WithTest(t))
+	l2Client := l2Network.inner.L2ELNode(match.FirstL2EL).EthClient()
+	l2tol1MessagePasser := bindings.NewBindings[bindings.L2ToL1MessagePasser](
+		bindings.WithClient(l2Client),
+		bindings.WithTo(predeploys.L2ToL1MessagePasserAddr),
+		bindings.WithTest(t))
+
+	addr, err := contractio.Read(l1Portal.DisputeGameFactoryAddr(), t.Ctx())
+	require.NoError(t, err, "Failed to read dispute game factory address")
```

Comment:
(optional) You can also use:
```suggestion
	addr := contract.Read(l1Portal.DisputeGameFactoryAddr())
```

Or:
```suggestion
	addr := l2Network.DisputeGameFactoryProxyAddr()
```

Reply:
`contract.Read` would create dependency cycles because the implementation references `dsl.EOA`.  But the l2Network method is a nice improvement.